src/geometry.F90

```diff
@@ -1582,5 +1614,327 @@ subroutine handle_lost_particle(p, message)
     end if
 
   end subroutine handle_lost_particle
+  
+!===============================================================================
+! CALC_OFFSETS calculates and stores the offsets in all fill cells. This
+! routine is called once upon initialization.
+!===============================================================================
+
+  subroutine calc_offsets(goal, map, univ, kounts, found)
+
+    integer, intent(in) :: goal                ! target universe ID
+    integer, intent(in) :: map                 ! map index in vector of maps
+    type(Universe), intent(in) :: univ         ! universe searching in
+    integer, intent(inout)     :: kounts(:,:)  ! Target count
+    logical, intent(inout)     :: found(:,:)   ! Target found
+
+    integer :: i                          ! index over cells
+    integer :: j, k, m                    ! indices in lattice
+    integer :: n                          ! number of cells to search
+    integer :: offset                     ! total offset for a given cell
+    integer :: cell_index                 ! index in cells array
+    type(Cell),     pointer :: c          ! pointer to current cell
+    type(Universe), pointer :: next_univ  ! next universe to cycle through
+    class(Lattice), pointer :: lat        ! pointer to current lattice
+
+    n = univ % n_cells
+    offset = 0
+
+    do i = 1, n
+      
+      cell_index = univ % cells(i)
+
+      ! get pointer to cell
+      c => cells(cell_index)
+      
+      ! ====================================================================
+      ! AT LOWEST UNIVERSE, TERMINATE SEARCH
+      if (c % type == CELL_NORMAL) then
+
+      ! ====================================================================
+      ! CELL CONTAINS LOWER UNIVERSE, RECURSIVELY FIND CELL
+      elseif (c % type == CELL_FILL) then
+        ! Set offset for the cell on this level
+        c % offset(map) = offset
+
+        ! Count contents of this cell
+        next_univ => universes(c % fill)
+        offset = offset + count_target(next_univ, kounts, found, goal, map)
+
+        ! Move into the next universe
+        next_univ => universes(c % fill)
+        c => cells(cell_index)
+
+      ! ====================================================================
+      ! CELL CONTAINS LATTICE, RECURSIVELY FIND CELL
+      elseif (c % type == CELL_LATTICE) then
+
+        ! Set current lattice
+        lat => lattices(c % fill) % obj
+
+        select type (lat)
+
+        type is (RectLattice)
+        
+          ! Loop over lattice coordinates
+          do j = 1, lat % n_cells(1)
+            do k = 1, lat % n_cells(2)
+              do m = 1, lat % n_cells(3)
+                lat % offset(map, j, k, m) = offset
+                next_univ => universes(lat % universes(j, k, m))
+                offset = offset+count_target(next_univ,kounts,found,goal,map)
```

Comment:
Procedure arguments should be separated by a [single space](http://mit-crpg.github.io/openmc/devguide/styleguide.html#whitespace-in-expressions)